node/src/components/linear_chain_sync.rs

```diff
@@ -144,6 +156,11 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
         let height = block_header.height();
         let hash = block_header.hash();
         trace!(%hash, %height, "downloaded linear chain block.");
+        if self.should_upgrade(block_header.era_id()) {
+            info!(era = block_header.era_id().0, "shutting down for upgrade");
+            self.stop_for_upgrade = true;
```

Comment:
Would it be cleaner to make this a new `State::StopForUpgrade` instead of a boolean?

Reply:
I followed the pattern from the `EraSupervisor` but you're right.

Reply:
Actually, this seems like a lot of code changes to only use for `linear_chain_sync` since it is all ignored in `linear_chain_fast_sync`.